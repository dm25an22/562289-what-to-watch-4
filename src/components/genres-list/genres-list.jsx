import React from "react";
import PropTypes from "prop-types";
import Genre from "../genre/genre.jsx";
import {connect} from "react-redux";
import {getUniqueGeners} from "../../reducer/data/selectors";
import {getCurrentGenre} from "../../reducer/app-state/selectors";
import {ActionCreator} from "../../reducer/app-state/app-state";

const GenresList = ({onGenreClick, currentGenre, genres}) => {
  return (
    <ul className="catalog__genres-list">
      {genres.map((it) => <Genre
        key={it}
        active={it === currentGenre}
        onGenreClick={onGenreClick}
        genre={it} />)}
    </ul>
  );
};

GenresList.propTypes = {
  currentGenre: PropTypes.string.isRequired,
  onGenreClick: PropTypes.func.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

// const mapStateToProps = (state) => {
//   return {
//     genres: getUniqueGeners(state),
//     currentGenre: getCurrentGenre(state),
//   };
// };

// const mapDispatchToProps = (dispatch) => {
//   return {
//     onGenreClick(genre) {
//       dispatch(ActionCreator.currentGenre(genre));
//     },
//   };
// };

// export {GenresList};
// export default connect(mapStateToProps, mapDispatchToProps)(GenresList);
export default GenresList;
