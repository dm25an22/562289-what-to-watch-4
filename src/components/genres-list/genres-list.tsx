import * as React from "react";
import Genre from "../genre/genre";

interface Props {
  onGenreClick: () => void,
  currentGenre: string,
  genres: string[]
}

const GenresList: React.FC<Props> = (props) => {
  const {
    onGenreClick,
    currentGenre,
    genres
  } = props;

  return (
    <ul className="catalog__genres-list">
      {genres.map((it) => <Genre
        key={it}
        active={it === currentGenre}
        onGenreClick={onGenreClick}
        genre={it} />)}
    </ul>
  );
};

export default GenresList;
