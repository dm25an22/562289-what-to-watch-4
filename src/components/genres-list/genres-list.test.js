import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list.jsx";

it(`Render GenreList`, () => {
  const tree = renderer.create(
      <GenresList
        currentGenre={`All genre`}
        onGenreClick={() => {}}
        genres={[
          `All genre`,
          `Action`,
          `Adventure`
        ]}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
