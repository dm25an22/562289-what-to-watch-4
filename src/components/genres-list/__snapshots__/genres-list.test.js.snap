// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenreList 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genre
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Action
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Adventure
    </a>
  </li>
</ul>
`;
